### DISCUSSION TITLE
Why did renovate stop offering me updates?

### DISCUSSION THREAD

**Discussion by ShlomoCode** [renovatebot/renovate#22733] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I installed renovate [in this project](https://github.com/ShlomoCode/yemot-router2/blob/master/renovate.json), it offered me some updates that I merged, but stopped offering new ones, even though [my package.json](https://github.com/ShlomoCode/yemot-router2/blob/master/package.json) has dependencies that need updating, for example eslint - from 8.39.0 to 8.42.0

### What you tried so far.

I tried to check "Check this box to trigger a request for Renovate to run again on this repository" in the Dependency Dashboard issue
No updates have been created

### Relevant debug logs

<details><summary>Logs</summary>

https://gist.github.com/ShlomoCode/bdbb26b7f4bb8900c8753d3e15acb7de
And here is the log of the initiated run that I created by checking the box in the Dependency Dashboard:
https://gist.github.com/ShlomoCode/91be269ccdaf08cca03303edef71d6ba

</details>

**Comment by rarkins** [comment]
Looking at the example dependency `eslint`:

```
          {
            "currentValue": "^8.34.0",
            "currentVersion": "8.42.0",
            "datasource": "npm",
            "depName": "eslint",
            "depType": "devDependencies",
            "homepage": "https://eslint.org",
            "lockedVersion": null,
            "packageName": "eslint",
            "prettyDepType": "devDependency",
            "registryUrl": "https://registry.npmjs.org",
            "sourceUrl": "https://github.com/eslint/eslint",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          },
```

It does not seem to have detected a locked version. It's possible that the lock file v3 is not being parsed properly

**Reply by rarkins** [reply]
Running locally produces the expected result:

```
               {
                 "depType": "devDependencies",
                 "depName": "eslint",
                 "currentValue": "^8.39.0",
                 "datasource": "npm",
                 "prettyDepType": "devDependency",
                 "lockedVersion": "8.39.0",
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "8.42.0",
                     "newValue": "^8.39.0",
                     "releaseTimestamp": "2023-06-02T23:47:40.126Z",
                     "newMajor": 8,
                     "newMinor": 42,
                     "updateType": "minor",
                     "isRange": true,
                     "isLockfileUpdate": true,
                     "branchName": "renovate/eslint-8.x-lockfile"
                   }
                 ],
                 "packageName": "eslint",
                 "warnings": [],
                 "versioning": "npm",
                 "sourceUrl": "https://github.com/eslint/eslint",
                 "registryUrl": "https://registry.npmjs.org",
                 "homepage": "https://eslint.org",
                 "currentVersion": "8.39.0",
                 "isSingleVersion": true,
                 "fixedVersion": "8.39.0"
               },
```

Do you mind adding a new commit to your default branch so that the existing cache will be discarded?

**Reply by ShlomoCode** [reply]
Yes, I recreated the package-lock.json file by deleting it and running npm install
https://github.com/ShlomoCode/yemot-router2/commit/68e08f39786d4097b25cee5bc1e19966b621550e
Here is a new log: https://gist.github.com/ShlomoCode/9d03665ccdad5cf721bf19a76c5eda4f

**Reply by rarkins** [reply]
Regenerating that lock file would have already "upgraded" most of the dependencies which were previously outedated.

From that log:

```
          {
            "depType": "devDependencies",
            "depName": "eslint",
            "currentValue": "^8.39.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "lockedVersion": null,
            "updates": [],
            "packageName": "eslint",
            "warnings": [],
            "versioning": "npm",
            "sourceUrl": "https://github.com/eslint/eslint",
            "registryUrl": "https://registry.npmjs.org",
            "homepage": "https://eslint.org",
            "currentVersion": "8.42.0"
          },
```

When I dry run the same repo locally:

```
               {
                 "depType": "devDependencies",
                 "depName": "eslint",
                 "currentValue": "^8.39.0",
                 "datasource": "npm",
                 "prettyDepType": "devDependency",
                 "lockedVersion": "8.42.0",
                 "updates": [],
                 "packageName": "eslint",
                 "warnings": [],
                 "versioning": "npm",
                 "sourceUrl": "https://github.com/eslint/eslint",
                 "registryUrl": "https://registry.npmjs.org",
                 "homepage": "https://eslint.org",
                 "currentVersion": "8.42.0",
                 "fixedVersion": "8.42.0"
               },
```

Although the result is the same, the log from my system includes `lockedVersion` and `fixedVersion` while from the hosted app it does not, which is surprising. Cc @nabeelsaabna @Gabriel-Ladzaretti

**Reply by nabeelsaabna** [reply]
@rarkins what Renovate version are you using locally ?
the hosted app is running `35.110.0`

**Reply by rarkins** [reply]
I'm running from source/main. Looks like it needs v35.113.0 or later: https://github.com/renovatebot/renovate/pull/22281

**Reply by ShlomoCode** [reply]
@rarkins I didn't fully understand, does this seem like a bug to you or did I set something wrong?

**Reply by rarkins** [reply]
It was on Renovate side, not your mistake

**Reply by Gabriel-Ladzaretti** [reply, marked as answer]
I assume this has been fixed by @nabeelsaabna , as it seems the hosted app is now at `v35.115.2`, and I do receive updates for a `package-lock.json`, which is at `lockfileVersion` 3.

https://github.com/Gabriel-Ladzaretti/lockfile-v3/pull/2/files

**Reply by ShlomoCode** [reply]
@Gabriel-Ladzaretti So why doesn't renovate create updates for me, for example for eslint?

**Reply by Gabriel-Ladzaretti** [reply]
https://github.com/ShlomoCode/yemot-router2/blob/68e08f39786d4097b25cee5bc1e19966b621550e/package-lock.json#L657-L658
https://www.npmjs.com/package/eslint

seems that you are at the latest version of `eslint`, so there is nothing to update to.

EDIT: i pasted the wrong permalink from the lockfile, [here](https://github.com/ShlomoCode/yemot-router2/blob/68e08f39786d4097b25cee5bc1e19966b621550e/package-lock.json#L2533-L2534) is the correct one, it is still the current latest tho :)

**Reply by ShlomoCode** [reply]
@Gabriel-Ladzaretti And why does renovate update a version only in `package-lock.json` and not in `package.json`?
https://github.com/ShlomoCode/yemot-router2/blob/c8a5d21523292662fec6336d599af2e88c6b6ac9/package.json#L24

**Reply by Gabriel-Ladzaretti** [reply]
Because you have a range version (`^8.39.0`) in your `package.json`, and the lock file has the actual installed version (`8.42.0`) that complies with this range.

https://semver.npmjs.com/
![image](https://github.com/renovatebot/renovate/assets/97394622/1edb65ae-a239-4e6f-850b-9a202e38bfa0)


you can pin your dependencies instead of relying on ranges
https://docs.renovatebot.com/dependency-pinning/#should-you-pin-your-javascript-dependencies

https://docs.renovatebot.com/configuration-options/#rangestrategy

**Reply by ShlomoCode** [reply]
@Gabriel-Ladzaretti Thanks for the explanation and the links!
I set rangeStrategy to bump in the configuration file and renovate did update eslint for me in `package.json`
But he did not offer me to update the `@types/node` package from `^20.2.5` to version `20.3.1`
What is the reason for this?
https://www.npmjs.com/package/@types/node

**Reply by Gabriel-Ladzaretti** [reply]
maybe because v20.3.1 was released just a few hours ago and renovate didn't scan you repo since.

**Reply by rarkins** [reply]
He's using the hosted app, so that also explains it

**Reply by Gabriel-Ladzaretti** [reply]
i was able to get updates for it in the hosted app a few minutes back tho
- https://github.com/Gabriel-Ladzaretti/lockfile-v3/pull/3

**Reply by rarkins** [reply]
Your case is _pinned_ though

**Reply by ShlomoCode** [reply]
> maybe because v20.3.1 was released just a few hours ago and renovate didn't scan you repo since.

Version 20.2.5 came out 19 days ago, and he didn't offer me to update it either...
And he did scan my database, because he immediately suggested that I update eslint to 8.42.0
https://github.com/ShlomoCode/yemot-router2/pull/14

**Reply by Gabriel-Ladzaretti** [reply]
not sure i understand, 
you initially installed @types/node v20.2.5 so you were at the latest at the time
https://github.com/ShlomoCode/yemot-router2/blame/a28502c547db96b15638693859a4653ac027e8be/package-lock.json#L155-L159

**Reply by ShlomoCode** [reply]
Oh, I got confused
But there is also 20.3.0 which was released 4 days ago
